### DISCUSSION TITLE
Help connecting external sensors

### DISCUSSION THREAD

**Discussion by Stew503** [osmandapp/OsmAnd#18956] [General]
Hello. I have purchased three different types of BLE temperature sensors and I can’t get any of them to be discovered by the OsmAnd external sensor pairing screen. I can connect all three with their own app but I can’t get them connected to OsmAnd. Any help would be greatly appreciated.

**Comment by tigrim** [comment]
Hello, what models of sensors do you use? I'll try to help you.

**Comment by Stew503** [comment]
Thank you for offering to help me. Here are links to the sensors I bought.

Just found this amazing item on AliExpress. Check it out! $7.49 50％ Off |
Tuya Smart Bluetooth BLE Temperature Humidity Thermometer Hygrometer
Detector for Greenhouse Wine Cellar Grow Tent Basement
https://a.aliexpress.com/_mt6yPaa

ThermoPro TP357 Digital Hygrometer Indoor Thermometer of 260FT, Bluetooth
Humidity Meter with Smart App, Room Humidity Gauge with Humidity Sensor,
2-Year Data Storage and Export https://a.co/d/5o1ruRB

Battery Monitor BM6 for Car RV 12V /24V Battery/Antigravity Batteries with
Bluetooth 4.0 Wireless Battery Sense Tester Battery Load Tester Compatible
with iPhone Apple Android Phone Free App… https://a.co/d/3SJzrUB



Thanks,
Chad

**Reply by tigrim** [reply]
Most smart device manufacturers make it so that their device only works with their native application. Technically, this is achievable, so these devices are not visible in other applications. At best, they can be seen, but it is not possible to obtain data from them. There is a high probability that the same applies to your temperature devices. If anything changes in the near future, I will let you know.
